### DISCUSSION TITLE
HTTP 400 Error with EtradeAccounts.list_acounts method

### DISCUSSION THREAD

**Discussion by gabrieldhofer** [jessecooper/pyetrade#90] [Q&A] [answered]
**Error Message:**
```
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://apisb.etrade.com/v1/accounts/list.json
```
**Code**
Trying to list accounts by following the docs [pyetrade accounts management docs](https://pyetrade.readthedocs.io/en/latest/examples.html#accounts-management):
```
  accounts = ETradeAccounts(
    consumer_key,
    consumer_secret,
    tokens['oauth_token'],
    tokens['oauth_token_secret'],
    dev=True
  )
  print(accounts.list_accounts(resp_format='json'))
```
Is anyone else experiencing something similar? Please let me know if I can provide more information about anything.
Thanks.

**Comment by kgstorm** [comment]
Did you have it working before or is this your first attempt?
if it’s your first attempt, are you making sure you’re using the right URL?  You are using the sandbox URL, so are you using your sandbox credentials or your real world credentials?

**Reply by gabrieldhofer** [reply]
Hi @kgstorm, thanks for replying.

This is my first attempt, and I'm using the sandbox credentials. 

I manually added a print statement in the **list_accounts** method to know for sure which URL it is using. 

It outputted the following url: `https://apisb.etrade.com/v1/accounts/list.json`

So, that makes sense since the `dev=True` parameter is passed to the **accounts** constructor.

btw, I am confident that I'm passing valid oauth_token and oauth_token_secret values to the **accounts** constructor as well.

**Reply by gabrieldhofer** [reply, marked as answer]
Okay, I found a solution.

I'm working in Windows WSL, and for some reason, there's a "clock skew" occurring (still not sure why).

But as a result, for the authorization step, I had to pass in my own "corrected" timestamp since my computer's timestamp was offset. 

So I passed in a corrected timestamp as the last parameter **my_timestamp**:

```
self.session = OAuth1Session(
    self.consumer_key,
    self.consumer_secret,
    callback_uri=self.callback_url,
    signature_type="AUTH_HEADER",

    timestamp=my_timestamp
)
```

Long story short, I just realized that another Oauth1 session is created in the constructor of the account class, so that constructor call needs the timestamp too, just as the authorization step required it (at least on my computer). 

Not sure if anyone else is experiencing clock skew but, if so, hopefully this can be of use.
